### PR TITLE
Remove old PythonFunction implementation 

### PR DESCRIPTION
**Pull request by banasraf**
#### Why we need this PR?
-Refactoring to remove redundant python function implementation. 

#### What happened in this PR?
 - What solution was applied:
     *All flavors of python function were implemented as wrappers around DLTensorPythonFunction operator. To achieve that, I wrote converting functions between DLPack tensor and numpy array. 
DLTensorPythonFunction was extended to support batch processing and per sample processing.*
 - Affected modules and functionalities:
     *[ Old python function operator was removed and the DLTensorPythonFunction was extended.]*
 - Key points relevant for the review:
     *[DLTensorPythonFunction implementation.  ]*
 - Validation and testing:
     *[ Additional tests for PythonFunction and TorchPython ]*
 - Documentation (including examples):
     *[ N.A. ]*

**JIRA TASK**: *[N.A.]*

### CODE REVIEW

**Comment by banasraf** [comment]
!build

**Comment by banasraf** [comment]
!build

File: dali/operators/python_function/dltensor_function.cc [hunk 1]
```diff
@@ -12,17 +12,23 @@
...
 #include "dali/operators/python_function/dltensor_function.h"
 #include "dali/pipeline/util/copy_with_stride.h"
 
 namespace dali {
 
 DALI_SCHEMA(DLTensorPythonFunctionImpl)
     .AddOptionalArg("synchronize_stream", "Synchronize CUDA stream", true)
+    .AddArg("function_id", R"code(Id of the python function)code", DALI_INT64)
+    .AddOptionalArg("num_outputs", R"code(Number of outputs)code", 1)
+    .AddArg("batch_processing", "Batch processing.", DALI_BOOL)
```
Comment (JanuszL) [inline dali/operators/python_function/dltensor_function.cc]: This relates only to CPU variant, right?

Comment (JanuszL) [reply dali/operators/python_function/dltensor_function.cc]: I see it now.

**Pull request merged**